src/libraries/Common/tests/System/Security/Cryptography/MLKemContractTests.cs

```diff
@@ -789,6 +789,115 @@ public static void ExportSubjectPublicKeyInfo_Disposed()
             Assert.Throws<ObjectDisposedException>(() => kem.ExportSubjectPublicKeyInfo());
         }
 
+        [Fact]
+        public static void TryExportPkcs8PrivateKey_EarlyExitForSmallBuffer()
+        {
+            MLKemContract kem = new(MLKemAlgorithm.MLKem512);
+            byte[] destination = new byte[85];
+            AssertExtensions.FalseExpression(kem.TryExportPkcs8PrivateKey(destination, out int written));
+            Assert.Equal(0, written);
+        }
+
+        [Fact]
+        public static void TryExportPkcs8PrivateKey()
+        {
+            int bufferSize = RandomNumberGenerator.GetInt32(87, 1024);
+            int writtenSize = RandomNumberGenerator.GetInt32(86, bufferSize);
+            bool success = (writtenSize & 1) == 1;
```

Comment:
OK... this took a bit of thought, but I see it.

* Make a random length buffer.
* Decide how many bytes were written, not exceeding the buffer.  (Because the upper bound is exclusive, it'll actually never say buffer.Length)
* Flip a coin (more or less) to decide if we return success.

So then the callback asserts it saw the fully sized buffer (no Slicing by the caller), reports the bytesWritten, and success (or not).

Now, back in the test method, assert that the things we wanted to return got returned.

Much better than where I started: "uh... why would Kevin expect that ML-KEM P8 exports are always odd lengthed?"